libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -98,7 +98,7 @@ void AC_Circle::init()
 /// set_circle_rate - set circle rate in degrees per second
 void AC_Circle::set_rate(float deg_per_sec)
 {
-    if (!is_equal(deg_per_sec,_rate)) {
+    if (!is_equal(deg_per_sec,(float)_rate)) {
```

Comment:
You can use "forwarding functions" to fix up types like AP_Param.with no code bloat. Here is an example. It is only one way to do it and probably out of date.. IOW the PR state can certainy be improved on, but seems like a good start as long as it is correct

```
#include <type_traits>

// A user defined type like AP_Param
template <typename T>
struct Param{
   Param(T t) : m_element{t}{}
   operator T ()const {return m_element;}
private:
   T m_element;
};

namespace std{

// create a specialisation of common_type for Param
// to return its runtime type
   template <typename T>
   struct common_type<::Param<T> >{
      typedef T type;
   };
}

// some function works only if both types arithmetic
// and returns the type of promotion of both
// look up SFINAE and enable_if for details
template <typename TL, typename TR>
typename std::enable_if<
    std::is_arithmetic<TL>::value && std::is_arithmetic<TR>::value
   , typename std::common_type<TL,TR>::type
>::type
fun(TL lhs, TR rhs)
{
   return lhs + rhs;
}

// if one or both isnt arithmetic
// then forward to the type of conversion
template <typename TL, typename TR>
typename std::enable_if<
   !(std::is_arithmetic<TL>::value && std::is_arithmetic<TR>::value)
   , decltype(fun(typename std::common_type<TL>::type{}, typename std::common_type<TR>::type{}))
>::type
fun(TL lhs, TR rhs)
{
  //todo  add static assert check that these are now arith etc
  typedef typename std::common_type<TL>::type lhs_type;
  typedef typename std::common_type<TR>::type rhs_type;
// forward to the arith version
  return fun(static_cast<lhs_type>(lhs),static_cast<rhs_type>(rhs));
}

#include <iostream>

int main()
{
    Param<int> x{1};
    Param<float> y{22.567};

    std::cout << static_cast<int>(x) <<'\n';
    std::cout << fun(1,2) <<'\n';
    std::cout << fun(x,24)  <<'\n';
    std::cout << fun(22.,y)  <<'\n';
    std::cout << fun(x,y)  <<'\n';
}

```